hazelcast/src/main/java/com/hazelcast/cp/internal/RaftGroupMembershipManager.java

```diff
@@ -658,7 +658,10 @@ private BiTuple<CPMember, CPGroupId> getEndpointWithMinLeaderships(Collection<CP
             CPGroupId groupId = null;
             int min = maxLeaderships;
             for (CPGroupSummary group : groups) {
-                for (CPMember member : group.members()) {
+                List<CPMember> groupMembers = new ArrayList<>(group.members());
+                // If the destination member is unavailable minimise the probability of using it next time
+                Collections.shuffle(groupMembers);
```

Comment:
Does it really help?

Reply:
It should. During the tests, I found that the leader constantly trying to be assigned to the same member. If this member was unavailable - it fails.

Reply:
ok

Reply:
Should it be backported to 5.3.z given how little impact the change has?

Reply:
Yes, I will create a backport.